File: app/src/main/java/com/example/weather1/views/Next5DaysFragment.java
package com.example.weather1.views;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.example.weather1.adapters.WeatherForecastAdapter;
import com.example.weather1.databinding.FragmentNext5DaysBinding;
import com.example.weather1.models.WeatherForecastResult;
import com.example.weather1.viewmodels.Next5DaysVM;


public class Next5DaysFragment extends Fragment {

    private FragmentNext5DaysBinding binding;
    WeatherForecastAdapter mAdapter;
    Next5DaysVM mNext5Days;




    public Next5DaysFragment() {
        // Required empty public constructor
    }


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;

    }



    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = FragmentNext5DaysBinding.inflate(inflater, container, false);
        View view = binding.getRoot();

        mNext5Days  = new ViewModelProvider(this).get(Next5DaysVM.class);
        mNext5Days.getNext5Days();

        configureRecyclerview();
        getForecastWeatherInfo();



        return view;
    }

    private void getForecastWeatherInfo() {
      mNext5Days.postsMutableLiveData.observe(getViewLifecycleOwner(), this::displayForecastWeather);

    }

    private void displayForecastWeather(WeatherForecastResult weatherForecastResult) {
        mAdapter  = new WeatherForecastAdapter(getContext(), weatherForecastResult);
        binding.recyclerForecast5d.setAdapter(mAdapter);

    }



    //--Configuration
    private void configureRecyclerview() {
        binding.recyclerForecast5d.setHasFixedSize(true);
        binding.recyclerForecast5d.setLayoutManager(new LinearLayoutManager(getContext(), LinearLayoutManager.VERTICAL, false));
    }


}

File: app/src/main/java/com/example/weather/Next5DaysFragment.kt
package com.example.weather

import android.os.Bundle
import android.view.*
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.fragment.app.activityViewModels
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.weather.databinding.FragmentNext24HoursBinding
import com.example.weather.databinding.FragmentNext5DaysBinding


class Next5DaysFragment : Fragment() {

    private lateinit var binding: FragmentNext5DaysBinding
    private val sharedViewModel: SharedViewModel by activityViewModels()

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        binding = FragmentNext5DaysBinding.inflate(inflater, container, false)
        setHasOptionsMenu(true)

        val adapter = DailyItemsAdapter()

        binding.recyclerView3.adapter = adapter
        val manager = LinearLayoutManager(context, RecyclerView.VERTICAL, false)
        binding.recyclerView3.layoutManager = manager


        sharedViewModel.status.observe(viewLifecycleOwner, { status ->
            when (status.name) {
                "LOADING" -> binding.nextDaysStatus.setImageResource(R.drawable.loading_animation)
                "DONE" -> {
                    binding.nextDaysStatus.visibility = View.GONE
                    binding.recyclerView3.visibility = View.VISIBLE
                }
                "ERROR" -> binding.nextDaysStatus.setImageResource(R.drawable.ic_connection_error)
            }
        })

        sharedViewModel.weather.observe(viewLifecycleOwner, { weatherResult ->
            if (weatherResult != null) {
                adapter.submitList(weatherResult.daily)
            }
        })

        return binding.root
    }


    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
        super.onCreateOptionsMenu(menu, inflater)
        inflater.inflate(R.menu.weather_menu, menu)
    }



}